### ISSUE TITLE
[Content] : Problems editing pages for Users with both FE and BE access

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#21249]
Reported via Support ticket: [105585](https://dotcms.zendesk.com/agent/tickets/105585)
Reproducible as of dotCMS 5.3.8.6, up to latest master (21.10)

When you edit a page in the dotCMS back-end, you can navigate to other pages in your Site -- actual HTML Pages or URL Mapped content -- and edit them right away. However, when your user has both `Back-End User` and `Front-End User` access, if you go to another HTML Page or URL Mapped content and try to edit it, you're editing the HTML Page you opened up from the **Site Browser** instead of the HTML Page or URL Mapped content you navigated to.

**To Reproduce**
1. Log into https://demo.dotcms.com/dotAdmin , go to the **Settings > Users** portlet, click the Admin User, un-check the `Front-End User` access box, and click **SAVE**.
2. Then, go to the **Site Browser**, and edit the `/index` page. The URL in the browser looks like this: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Findex&language_id=1&host_id=48190c8c-42c4-46af-8d1a-0cd5db894797
3. On the page's main nav, click **DESTINATIONS**. The URL in the browser now changes to this, which is correct: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Fdestinations%2Findex
4. Click the **EDIT** button to edit the current page. What you're now editing are the contents of the `/destinations/index` page. This is the expected behavior.
5. Now, go back to the **Settings > Users** portlet, click the Admin User, check the `Front-End User` access box, and click **SAVE**.
6. Go back to the **Site Browser**, and edit the `/index` page again. This is the URL in the browser: https://demo.dotcms.com/dotAdmin/#/edit-page/content?url=%2Findex&language_id=1&host_id=48190c8c-42c4-46af-8d1a-0cd5db894797
7. On the page's main nav, click **DESTINATIONS**. Notice that the URL in the browser **did not change at all this time**.
8. Click the **EDIT** button to edit the current page.
9. You're editing the contents of the `/index` page instead of the `/destinations/index` page, which is the page you clicked last time. This is **NOT** the expected behavior.

**Expected behavior**
Back-end Users should be able to click any link and edit the contents they see without problems.

**Additional context**
This issue seems to be related to the code changes from #20452 . After debugging the code with @freddyucv , we believe that the fix is to replace this line:
https://github.com/dotCMS/core/blob/40217a9e09aecb2715c97cbb2ecc3cd1c133b366/dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java#L58
With this one:
```java
final PageMode mode = userApi.getLoggedInUser(request) != null ? PageMode.getWithNavigateMode(request) : PageMode.setPageMode(request, PageMode.LIVE);
```
The only workaround we have found so far is to uncheck the `Front-End User` access checkbox for the conflicting Users.

**Comment by jcastro-dotcms** [comment]
Fix verification scenario provided by Will:

- Create a test HTML Page.
- Log in as a FE user.
- Try to request such a page in working state.
- If data is returned, then the proposed fix is not correct.

I haven't been able to test this as latest master is currently not allowing users to add content to pages

**Comment by jcastro-dotcms** [comment]
Another customer experiencing this problem: https://dotcms.zendesk.com/agent/tickets/106292

**Comment by jdotcms** [comment]
The changes are working as an expected on both scenarios

**Comment by wezell** [comment]
PR: #21667

**Comment by bryanboza** [comment]
Fixed, unable to reproduce on release-22.03 // Docker // FF
